src/DateTime/DateAdjusted.ts

```diff
@@ -0,0 +1,18 @@
+import type { Moment } from "moment";
+import { getSettings } from "../Config/Settings";
+
+/**
+ * Returns the current moment, adjusted for the "Next day start hour" setting.
+ *
+ * If the current time is before the "next day start hour" (e.g., 2 AM),
+ * this function will return a moment object for the end of the *previous* day.
+ * Otherwise, it returns the current, unaltered moment.
+ */
+export function momentAdjusted(): Moment {
+    const { nextDayStartHour } = getSettings();
+    const now = window.moment();
+    if (now.hour() < nextDayStartHour) {
```

Comment:
What happens when clocks change? There needs to be tests for this - including in timezones close to the international date line...

I have written up the results of my previous experiments in timezones in tests:

https://publish.obsidian.md/tasks-contributing/Testing/Testing+and+Time+Zones